src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -104,93 +112,45 @@ public override async Task<IEnumerable<string>> DownloadVersion(PackageURL purl,
+                
+                string? targetName = $"nuget-{packageName}@{packageVersion}";
+                string extractionPath = Path.Combine(TopLevelExtractionDirectory, targetName);
+                if (doExtract && Directory.Exists(extractionPath) && cached)
                 {
-                    if (catalogPage.TryGetProperty("items", out JsonElement itemElement))
-                    {
-                        foreach (JsonElement item in itemElement.EnumerateArray())
-                        {
-                            JsonElement catalogEntry = item.GetProperty("catalogEntry");
-                            string? version = catalogEntry.GetProperty("version").GetString();
-                            if (version != null && version.Equals(packageVersion, StringComparison.InvariantCultureIgnoreCase))
-                            {
-                                string? archive = catalogEntry.GetProperty("packageContent").GetString();
-                                HttpResponseMessage? result = await httpClient.GetAsync(archive);
-                                result.EnsureSuccessStatusCode();
-                                Logger.Debug("Downloading {0}...", purl?.ToString());
-
-                                string? targetName = $"nuget-{packageName}@{packageVersion}";
-                                string extractionPath = Path.Combine(TopLevelExtractionDirectory, targetName);
-                                if (doExtract && Directory.Exists(extractionPath) && cached == true)
-                                {
-                                    downloadedPaths.Add(extractionPath);
-                                    return downloadedPaths;
-                                }
+                    downloadedPaths.Add(extractionPath);
+                    return downloadedPaths;
+                }
 
-                                if (doExtract)
-                                {
-                                    downloadedPaths.Add(await ExtractArchive(targetName, await result.Content.ReadAsByteArrayAsync(), cached));
-                                }
-                                else
-                                {
-                                    targetName += Path.GetExtension(archive) ?? "";
-                                    await File.WriteAllBytesAsync(targetName, await result.Content.ReadAsByteArrayAsync());
-                                    downloadedPaths.Add(targetName);
-                                }
-                                return downloadedPaths;
-                            }
-                        }
+                
+                if (doExtract)
+                {
+                    IEnumerable<string>? files = await downloader.CoreReader.GetFilesAsync(extractionPath, cancellationToken);
+                    if (files is not null && files.Any())
+                    {
+                        downloadedPaths.Add(extractionPath);
```

Comment:
I like that after these changes, this method is much smaller, but I wonder whether we can do even better - what do you think about splitting the extract and non-extract into two separate private helper methods, which this method can call? Might help make the high-level logic easier to read, rather than having long method bodies.

Reply:
I can do that!

Reply:
Actually, @pmalmsten don't know what version of the PR you were reviewing when looking at this, but the logic for `doExtract` and the else is now only 10 lines long. Let me know if you think it's still worth refactoring it.